lucille-core/src/main/java/com/kmwllc/lucille/stage/ParseFloats.java

```diff
@@ -23,8 +26,14 @@ public ParseFloats(Config config) {
   public Iterator<Document> processDocument(Document doc) throws StageException {
     if (doc.has(field)) {
       String value = doc.getString(this.field);
-      List<Float> floats = Arrays.stream(value.replaceAll("\\[", "").replaceAll("\\]", "").replaceAll(",", "").split(" "))
-          .map(v -> Float.parseFloat(v)).collect(Collectors.toList());
+      List<Float> floats;
+      try {
+        floats = Arrays.stream(value.replaceAll("\\[", "").replaceAll("\\]", "").replaceAll(",", "").split(" "))
```

Comment:
I suspect that using Jackson to parse the string as json will be faster than applying these four regexes and creating new strings each time. The error reporting from jackson will probably also be better. Here is how it can be done:

```
    ObjectMapper mapper = new ObjectMapper(); // should be private static final
    String floatString = "[1.2, 3.4, 5.6]";
    Float[] floats = mapper.readValue(floatString, Float[].class); // if we want an array
    // IntelliJ will say TypeReference<List<Float>> can be TypeReference<> but be explicit so jackson can check types for us
    List<Float> floatList = mapper.readValue(floatString, new TypeReference<List<Float>>(){});
```

I know that this PR was meant for tests only
Up to you whether to asses performance of the jackson approach and do it here in this PR, or create a ticket to do it in a followup PR

Reply:
i changed it